internal/nvmeof/nodeserver/nodeserver.go

```
@@ -0,0 +1,631 @@
+
+			return
+		}
+	}
+
+	// remove the file/directory created on staging path
+	if transaction.isStagePathCreated {
+		err = os.Remove(stagingTargetPath)
+		if err != nil {
+			log.ErrorLog(ctx, "failed to remove stagingtargetPath: %s with error: %v", stagingTargetPath, err)
+			// continue on failure to disconnect the image
+		}
+	}
+}
+
+// createTargetMountPath check if the mountPath already has something mounted
+// on it. If not, the directory (for a filesystem volume) will be created.
+//
+// This function returns 'true' in case there is something mounted on the given
+// path already, 'false' when the path exists, but nothing is mounted there.
+func (ns *NodeServer) createTargetMountPath(ctx context.Context, mountPath string, isBlock bool) (bool, error) {
+	isMnt, err := ns.Mounter.IsMountPoint(mountPath)
+	if err == nil {
+		return isMnt, nil
+	}
+	if !os.IsNotExist(err) {
+		return false, fmt.Errorf("path %q exists, but detecting it as mount point failed: %w", mountPath, err)
+	}
+
+	// filesystem volume needs a directory
+	if !isBlock {
+		// Create a mountpath directory
+		if err = util.CreateMountPoint(mountPath); err != nil {
+			return false, fmt.Errorf("failed to create mount path %q: %w", mountPath, err)
+		}
+
+		return false, nil
+	}
+
+	// block volume checks
+	// #nosec
+	pathFile, err := os.OpenFile(mountPath, os.O_CREATE|os.O_RDWR, 0o750)
+	if err != nil {
+		log.DebugLog(ctx, "Failed to create mountPath:%s with error: %v", mountPath, err)
+
+		return false, fmt.Errorf("failed to create mount file %q: %w", mountPath, err)
+	}
+	if err = pathFile.Close(); err != nil {
+		log.DebugLog(ctx, "Failed to close mountPath:%s with error: %v", mountPath, err)
+
+		return false, fmt.Errorf("failed to close mount file %q: %w", mountPath, err)
+	}
+
+	return false, nil
+}
+
+// isNotMountPoint checks whether MountPoint does not exists and
+// also discards error indicating mountPoint exists.
```

`also discards error indicating mountPoint does not exists`

---

fixed